modules/questions.py

```diff
@@ -337,48 +357,142 @@ async def cb_post_questions(
             why=why,
         )
 
-    async def post_random_oldest_question(self, _event_type) -> None:
-        """Post random oldest not started question.
-        Triggered automatically six hours after non-posting any question
-        (unless the last was already posted automatically using this method).
+    def is_time_for_autopost_not_started(self) -> bool:
+        return (
+            self.last_not_started_autopost_attempt_dt
+            < datetime.now() - self.not_started_question_autopost_interval
+        )
+
+    async def last_msg_in_general_was_autoposted(self) -> bool:
+        channel = cast(
+            TextChannel, self.utils.client.get_channel(int(general_channel_id))
+        )
+        async for msg in channel.history(limit=1):
+            if msg.content.startswith(self.AUTOPOST_NOT_STARTED_MSG_PREFIX):
+                return True
+        return False
+
+    async def autopost_not_started(self) -> Response:
+        """Choose a random question from the oldest not started questions and post to `#general` channel
+
+        Returns `Response` for ease of debugging with callbacks.
         """
-        # get channel #general
-        channel = cast(Thread, self.utils.client.get_channel(int(general_channel_id)))
+        current_time = datetime.now()
+        self.last_not_started_autopost_attempt_dt = current_time
 
-        # query for questions with status "Not started" and not tagged as "Stampy"
-        questions_df_filtered = self.coda_api.questions_df.query(
-            "status == 'Not started'"
+        if await self.last_msg_in_general_was_autoposted():
+            self.log.info(
+                self.class_name,
+                msg="Last message in #general was an autoposted question with status `Not started` -> skipping autoposting",
+            )
+            return Response(confidence=10)
```

Comment:
I'm sure this is solving some currently existing problem, but when I see this code, "ease of debugging" is NOT what comes to my mind,, it's very puzzling to me what this is supposed to do

can we use something with a name that spells out WTF returning an empty Response with highest confidence is a good idea and call that thing from here as
`return the_thing_that_explains_what_is_happening_in_the_name`, perhaps?

Reply:
I actually ~~found~~ reminded myself of a better way to debug these, so I will just remove that part